iml-agent/src/action_plugins/postoffice.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{agent_error::ImlAgentError, env};
+
+use std::collections::HashSet;
+use tokio::{fs, io::AsyncWriteExt};
+
+async fn write_config(mut routes: HashSet<String>) -> Result<(), ImlAgentError> {
+    let conf_file = env::get_var("POSTMAN_CONF_PATH");
+    let mut file = fs::File::create(conf_file).await?;
+    let rt: Vec<String> = routes.drain().collect();
```

Comment:
```suggestion
    let rt: Vec<String> = Vec::from_iter(routes);
```

I think that is a little clearer and also allows the `routes` to not be mutable.